bin/npg_publish_tree.pl

```diff
@@ -127,6 +129,21 @@ sub _make_filter_fn {
   };
 }
 
+sub _read_metadata_stdin {
```

Comment:
The two functions and mostly identical, so the common parts should be in a third function.